src/app/public/theming/theme-if.directive.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  Directive,
+  Host,
+  Input,
+  OnDestroy,
+  Optional,
+  TemplateRef,
+  ViewContainerRef
+} from '@angular/core';
+
+import {
+  Subject
+} from 'rxjs';
+
+import {
+  takeUntil
+} from 'rxjs/operators';
+
+import {
+  SkyThemeSettings
+} from './theme-settings';
+
+import {
+  SkyThemeService
+} from './theme.service';
+
+import {
+  SkyThemeDirective
+} from './theme.directive';
+
+/**
+ * Component that works like `ngIf` to show markup for matching theme.
+ *
+ * If the directive is within a `skyTheme` directive, it uses settings from that directive.
+ */
+@Directive({
+  selector: '[skyThemeIf]'
+})
+export class SkyThemeIfDirective implements OnDestroy {
+  private context: string;
+  private currentTheme: SkyThemeSettings | undefined;
+  private ngUnsubscribe = new Subject();
+  private hasView: boolean = false;
```

Comment:
Since you're specifying an initial value for `hasView`, you can let type inference do its thing and leave off the explicit type declaration.